test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1091,10 +1091,15 @@ private synchronized void reset(boolean wipeData) throws IOException {
         final int defaultMinMasterNodes = (numberOfMasterNodes / 2) + 1;
         final List<NodeAndClient> toStartAndPublish = new ArrayList<>(); // we want to start nodes in one go due to min master nodes
         final Runnable onTransportServiceStarted = () -> rebuildUnicastHostFiles(toStartAndPublish);
+        boolean bootstrapNodeRequired = prevNodeCount == 0;
         for (int i = 0; i < numSharedDedicatedMasterNodes; i++) {
             final Settings.Builder settings = Settings.builder();
             settings.put(Node.NODE_MASTER_SETTING.getKey(), true);
             settings.put(Node.NODE_DATA_SETTING.getKey(), false);
+            if (bootstrapNodeRequired) {
+                settings.put(INITIAL_MASTER_NODE_COUNT_SETTING.getKey(), numSharedDedicatedMasterNodes);
+                bootstrapNodeRequired = false;
```

Comment:
why only run the bootstrapping on a single node?

Reply:
It's sufficient and more obviously correct, but either way is fine by me. Running it on multiple nodes requires `validateClusterFormed()` before the next node starts up, and this isn't especially clear. Your call.

Reply:
We discussed this on another channel, and decided to only do the auto-bootstrapping when autoManageMinMaster mode is active. We also decided to have multiple nodes participate in / run the bootstrapping process.

Reply:
Done in 150f47b6a50b434fd4d9569683508cdc1a3f5d7d